src/Surfnet/StepupMiddleware/CommandHandlingBundle/Tests/Pipeline/AuthorzingStageTest.php

```diff
@@ -0,0 +1,136 @@
+<?php
+
+/**
+ * Copyright 2014 SURFnet bv
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Surfnet\StepupMiddleware\CommandHandlingBundle\Tests\Pipeline;
+
+use Mockery as m;
+use PHPUnit_Framework_TestCase as UnitTest;
+use Psr\Log\NullLogger;
+use Surfnet\StepupMiddleware\CommandHandlingBundle\Pipeline\AuthorizingStage;
+
+class AuthorizingStageTest extends UnitTest
+{
+    /**
+     * @var \Mockery\MockInterface mock of Symfony\Component\Security\Core\Authorization\AuthorizationCheckerInterface
+     */
+    private $authorizationChecker;
+
+    /**
+     * @var NullLogger
+     */
+    private $logger;
+
+    public function setUp()
+    {
+        $this->logger = new NullLogger();
+        $this->authorizationChecker = m::mock(
+            'Symfony\Component\Security\Core\Authorization\AuthorizationCheckerInterface'
+        );
+    }
+
+    /**
+     * @test
+     * @group pipeline
+     */
+    public function when_a_command_has_no_marker_interface_authorization_is_granted_by_default()
+    {
+        $command = m::mock('Surfnet\StepupMiddleware\CommandHandlingBundle\Command\Command');
+        $this->authorizationChecker->shouldReceive('isGranted')->never();
+
+        $authorizingStage = new AuthorizingStage($this->logger, $this->authorizationChecker);
+
+        $authorizingStage->process($command);
+    }
+
+    /**
+     * @test
+     * @group pipeline
+     * @dataProvider interfaceToRoleMappingProvider
+     *
+     * @param string $interface
+     * @param string $role
+     */
+    public function a_command_with_a_marker_interface_triggers_a_check_for_the_correct_role($interface, $role)
+    {
+        $command = m::mock('Surfnet\StepupMiddleware\CommandHandlingBundle\Command\Command, ' . $interface);
+        $this->authorizationChecker
+            ->shouldReceive('isGranted')
+            ->once()
+            ->with([$role])
+            ->andReturn(true);
+
+        $authorizingStage = new AuthorizingStage($this->logger, $this->authorizationChecker);
+
+        $authorizingStage->process($command);
+    }
+
+    /**
+     * @test
+     * @group pipeline
+     */
+    public function when_a_command_implements_multiple_marker_interfaces_at_least_one_corresponding_role_is_required()
+    {
+        //@todo discuss: this sequence is tightly coupled to the sequence of ifs in the stage itself,
+        //               making for a brittle test... thoughts?
```

Comment:
It ensures that it works as `atLeastOne` indicates: having one role that is required is enough, I think it is valuable to test that you are required to have at least one role, not all roles that are allowed to execute the command. 
Just removing the counting expectations ('once()', 'never()') would remove the brittleness while leaving the test itself intact. Thoughts?